### PR TITLE
Gather stats from Strongswan VPN server

### PR DESCRIPTION
**Pull request by TuxPowered42**

### CODE REVIEW

File: igcollect/strongswan.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    # ---- >8 ----
+
+    # Let's say that the line containing word 'remote' and IP address of the client
+    # identifies a single client.
+    line_re = re.compile(r'^\s+remote .* @')
+
+
+    for line in check_output(['/usr/sbin/swanctl', '--list-sas']).splitlines():
+        if line_re.match(line.decode()):
+            clients += 1
```
Comment (lamaral) [inline igcollect/strongswan.py]: I think you can replace this for loop with a `line_re.finditer()`, if you add `re.MULTILINE` to the compile arguments.

File: igcollect/strongswan.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    #     local  0.0.0.0/0 ::/0
+    #     remote XXX.XXX.XXX.XXX/32 XXX:XXX:XXX::XXX/128
+    # ---- >8 ----
+
+    # Let's say that the line containing word 'remote' and IP address of the client
+    # identifies a single client.
+    line_re = re.compile(r'^\s+remote .* @')
+
+
+    for line in check_output(['/usr/sbin/swanctl', '--list-sas']).splitlines():
```
Comment (TheFRedFox) [inline igcollect/strongswan.py]: I would suggest to write it as that:
``` python
output = check_output(...).decode()
lines = output.splitlines()
clients = sum(1 for l in lines if line_re.match(l))
```

File: igcollect/strongswan.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    #     installed 3354s ago, rekeying in 53s, expires in 606s
+    #     in  42424242 (-|0x000675ec), 9577036 bytes, 63405 packets,     2s ago
+    #     out 42424242 (-|0x000675ec), 20466058 bytes, 68523 packets,     2s ago
+    #     local  0.0.0.0/0 ::/0
+    #     remote XXX.XXX.XXX.XXX/32 XXX:XXX:XXX::XXX/128
+    # ---- >8 ----
+
+    # Let's say that the line containing word 'remote' and IP address of the client
+    # identifies a single client.
+    line_re = re.compile(r'^\s+remote .* @')
```
Comment (TheFRedFox) [inline igcollect/strongswan.py]: I do it correctly, that you just want to match the 4th line, right? Not the 12th line.

File: igcollect/strongswan.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+
+def count_connected_clients():
+    clients = 0
+    
+    # Single connected users looks like this:
+
+    # ---- 8< ----
+    # vpn_by_id_group_xyz: #974, ESTABLISHED, IKEv2, 4242424242424242_i 4242424242424242_r*
+    #   local  'XXXXXX.innogames.de' @ XXX.XXX.XXX.XXX[4500]
```
Comment (TheFRedFox) [inline igcollect/strongswan.py]: Maybe rather go for `'XXXXX.example.com'` for the example?

File: igcollect/strongswan.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env python
+"""igcollect - Strongswan stats
+
+Copyright (c) 2020 InnoGames GmbH
+"""
+
+from argparse import ArgumentParser
+from subprocess import check_output
+from time import time
```
Comment (TheFRedFox) [inline igcollect/strongswan.py]: You are shadowing the module `time` with that import. For me it is fine, but in the future one needs to think about it.

File: igcollect/strongswan.py [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+def parse_args():
+    parser = ArgumentParser()
+    parser.add_argument('--prefix', default='strongswan')
+    return parser.parse_args()
+
+
+def main():
+    args = parse_args()
+    template = args.prefix + '.{} {} ' + str(int(time()))
+    print(template.format('clients', count_connected_clients()))
```
Comment (TheFRedFox) [inline igcollect/strongswan.py]: If `clients` is fixed, we could also already put it in the template.

**Comment by Woellchen** [comment]
@TuxPowered42 any progress?

File: igcollect/strongswan.py (hunk 4 above)
Comment (TuxPowered42) [reply igcollect/strongswan.py:32]: Fixed.

File: igcollect/strongswan.py (hunk 6 above)
Comment (TuxPowered42) [reply igcollect/strongswan.py:22]: I expect more stats to be added in future.

File: igcollect/strongswan.py (hunk 2 above)
Comment (TuxPowered42) [reply igcollect/strongswan.py:49]: Fixed.

File: igcollect/strongswan.py (hunk 3 above)
Comment (TuxPowered42) [reply igcollect/strongswan.py:46]: Line 12 has no `@` sign in it, which is matched by this regexp.

File: igcollect/strongswan.py (hunk 1 above)
Comment (TuxPowered42) [reply igcollect/strongswan.py:51]: I rewrote it the way @TheFRedFox suggested, sorry.

File: igcollect/strongswan.py (hunk 5 above)
Comment (TuxPowered42) [reply igcollect/strongswan.py:9]: As discussed on Slack, I consider shadowing good for this simple script.

**Pull request merged**